src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -1431,7 +1431,8 @@ private static LambdaConversionResult IsAnonymousFunctionCompatibleWithDelegate(
                 return LambdaConversionResult.BadTargetType;
             }
 
-            if (anonymousFunction.HasExplicitReturnType(out var refKind, out var returnType))
+            bool hasExplicitReturnType;
+            if (hasExplicitReturnType = anonymousFunction.HasExplicitReturnType(out var refKind, out var returnType))
```

Comment:
```suggestion
            bool hasExplicitReturnType = anonymousFunction.HasExplicitReturnType(out var refKind, out var returnType);
            if (hasExplicitReturnType)
```

Generally try to avoid assignment in `if` conditionals.

Reply:
I think that would lose nullability information like `[NotNullWhen(true)]` - but that's not used here.